libvirt/README.md

```diff
@@ -44,3 +44,286 @@ If you want to set default password for podvm debugging then you can use guestfi
+|  Cluster  |   Type  |    Plan   |                  Vms                  |
++-----------+---------+-----------+---------------------------------------+
+| peer-pods | generic | peer-pods | peer-pods-master-0,peer-pods-worker-0 |
++-----------+---------+-----------+---------------------------------------+
+$ kubectl get nodes
+NAME                 STATUS   ROLES                  AGE     VERSION
+peer-pods-master-0   Ready    control-plane,master   6m8s    v1.25.3
+peer-pods-worker-0   Ready    worker                 2m47s   v1.25.3
+```
+
+## Prepare the Pod VM volume
+
+In order to build the Pod VM without installing the build tools, you can use the Dockerfiles hosted on [../podvm](../podvm) directory to run the entire process inside a container.
+
+To build a Ubuntu VM image, run:
+
+```
+$ cd ../podvm
+$ docker build --no-cache -t podvm_builder -f Dockerfile.podvm_builder .
+$ docker build --no-cache -t podvm_libvirt --build-arg BUILDER_IMG=podvm_builder:latest \
+	--build-arg CLOUD_PROVIDER=libvirt -f Dockerfile.podvm .
+```
+
+The qcow2 image file will be created inside the `podvm_libvirt` image and can be extracted as shown below:
+
+```
+$ podvm_container=$(docker create podvm_libvirt:latest /bin/bash)
+$ file_name=$(docker export ${podvm_container} | tar -t | grep podvm)
+$ docker cp ${podvm_container}:${file_name} podvm.qcow2
+$ docker rm ${podvm_container}
+```
+
+Next you will need to create a volume on libvirt's system storage and upload the image content. That volume is used by
+the cloud-api-adaptor program to instantiate a new Pod VM. Still on the `image` directory, you should run the following
+commands:
+
+```
+$ export IMAGE=<full-path-to-qcow2>
+
+$ virsh vol-create-as --pool default --name podvm-base.qcow2 --capacity 20G --allocation 2G --prealloc-metadata --format qcow2
+$ virsh vol-upload --vol podvm-base.qcow2 $IMAGE --pool default --sparse
+```
+
+You should see that the `podvm-base.qcow2` volume was properly created:
+
+```
+$ virsh -c qemu:///system vol-info --pool default podvm-base.qcow2
+Name:           podvm-base.qcow2
+Type:           file
+Capacity:       6.00 GiB
+Allocation:     631.52 MiB
+```
+
+## Install and configure Confidential Containers and cloud-api-adaptor in the cluster
+
+The easiest way to install the cloud-api-adaptor along with Confidential Containers in the cluster is through the
+Kubernetes operator available in the `install` directory of this repository.
+
+Start by creating a public/private RSA key pair that will be used by the cloud-api-provider program, running on the
+cluster workers, to connect with your local libvirtd instance without password authentication. Assume you are in the
+`libvirt` directory, do:
+
+```
+$ cd ../install/overlays/libvirt
+$ ssh-keygen -f ./id_rsa -N ""
+$ cat id_rsa.pub >> ~/.ssh/authorized_keys
+```
+**Note**: ensure that `~/.ssh/authorized_keys` has the right permissions (read/write for the user only) otherwise the
```

Comment:
comment can be removed now?

Reply:
I didn't get. Why remove it? Mind to expand a little bit on your comment?

Reply:
you do chmod no?